scripts/istio-utils.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+# A set of utility functions for working with Istio
+# Calls to kubectl expects KUBECONFIG environment variable to be set correctly.
+
+
+script_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+root_dir=$(dirname "${script_dir}")
+output_dir="${root_dir}"/_output
+sample_dir="${root_dir}"/istio//samples
+
+
+# get_istio_version attempts to parse the istio version from Gopkg.toml if not set.
+# If parsing fails, default to setting to the latest version.
+function get_istio_version() {
+    if [[ "x${ISTIO_VERSION}" = "x" ]] ; then
+        # not using regex here because of some compatibility issues across different OS
+        ISTIO_VERSION=$(grep -A 1 "istio.io/istio" ${root_dir}/Gopkg.toml | grep "version" | cut -d'"' -f 2 | tr -d "=<>~^")
+    fi
+
+    if [[ "x${ISTIO_VERSION}" = "x" ]] ; then
+      echo "Unable to parse Istio version from dependencies - attempting to use latest version"
+      ISTIO_VERSION=$(curl -L -s https://api.github.com/repos/istio/istio/releases/latest | \
+                      grep tag_name | sed "s/ *\"tag_name\": *\"\\(.*\\)\",*/\\1/")
+    fi
+
+    if [[ "x${ISTIO_VERSION}" = "x" ]] ; then
+      printf "Unable to get latest Istio version. Set ISTIO_VERSION env var and re-run. For example: export ISTIO_VERSION=1.1.11"
+      exit;
+    fi
+
+    printf ${ISTIO_VERSION}
+}
+
+
+# get_istio downloads specified Istio source
+function get_istio() {
+    version=$(get_istio_version)
+
+    OS="$(uname)"
+    if [[ "x${OS}" = "xDarwin" ]] ; then
+      OSEXT="osx"
+    else
+      OSEXT="linux"
+    fi
+
+    NAME="istio-${version}"
+    URL="https://github.com/istio/istio/releases/download/${version}/istio-${version}-${OSEXT}.tar.gz"
+    if ! [[ -d ${root_dir}/_output/${NAME} ]]; then
+      printf "Downloading %s from %s ..." "$NAME" "$URL"
+      cd ${root_dir}/_output && curl -L "$URL" | tar xz
```

Comment:
this should reference the `output_dir` variable - also, it should `mkdir -p "${output_dir}"` to ensure the directory exists